test/OpenIddict.Validation.IntegrationTests/OpenIddictValidationIntegrationTests.cs

```diff
@@ -63,6 +63,34 @@ public async Task ProcessAuthentication_EvalutesCorrectValidatedTokens()
         Assert.Equal(0, response.Count);
     }
 
+    [Fact]
+    public async Task ProcessAuthentication_DoesNotResolveServerConfigurationWhenAccessTokenIsMissing()
+    {
+        // Arrange
+        await using var server = await CreateServerAsync(options =>
+        {
+            options.AddEventHandler<ProcessAuthenticationContext>(builder =>
+            {
+                builder.UseInlineHandler(context =>
+                {
+                    // Assert
+                    Assert.True(context.IsRejected);
```

Comment:
Are we sure this inline handler is invoked/this assert is reached? When a handler rejects the request, the next handler in the pipeline is not invoked, so this shouldn't be possible do such a check here.

Reply:
You are right, I refactored the test to verify that the demand is already rejected before resolving server configuration when access token is missing. I hope this one makes more sense and if not, I would be happy to remove the test altogether or follow a more specific suggestion.

Reply:
I don't think it will work (just like the test under the one you wrote, `ProcessAuthentication_RejectsDemandWhenAccessTokenIsMissing`, is flaky and has an inline handler that is not executed: since the previous handler marks the request as rejected, it's never invoked).

Let's remove it for now: I'll add a test that creates a mocked `IConfigurationManager<T>` that throws when calling `GetConfigurationAsync()` in a separate PR (unless you want to do it, of course 😃)

Reply:
Something like this should do the trick 😃 

```csharp
[Fact]
public async Task ProcessAuthentication_RejectsDemandWithoutResolvingServerConfigurationWhenAccessTokenIsMissing()
{
    // Arrange
    var manager = Mock.Of<IConfigurationManager<OpenIddictConfiguration>>();

    await using var server = await CreateServerAsync(options =>
    {
        options.Configure(options => options.ConfigurationManager = manager);
    });

    await using var client = await server.CreateClientAsync();

    // Act
    var response = await client.PostAsync("/authenticate", new OpenIddictRequest());

    // Assert
    Assert.Equal(0, response.Count);
    Mock.Get(manager).Verify(manager => manager.GetConfigurationAsync(It.IsAny<CancellationToken>()), Times.Never());
}
```

Reply:
Thanks Kevin! :) You are of course right again, the updated test also wasn't good. I removed the test completely and if it is all right with you, I would gladly accept your offer and leave creating the proper test to you :)